### PR TITLE
feat: investor judging

### PR DESCRIPTION
**Pull request by cyu60**
# Pull Request: Implement Flexible Judging System for Demo Day

## Overview
This PR implements a customizable judging system that supports both traditional numeric scoring and investment-focused binary decisions. This enables MentorMate to better serve different event formats, particularly Stanford Demo Day which uses an investment-oriented approach.

### 1. Dual Judging Mode Support
- Added `JudgingMode` enum with `Traditional` and `Investment` options
- Created track-specific judging mode configuration
- Implemented conditional interface rendering based on track's judging mode

### 2. Investment Decision Interface
- Built a choice-based investment interface with `invest`/`maybe`/`pass` options
- Added interest level slider (0-5) for investment decisions
- Enhanced the judge dashboard to properly display investment decisions

### 3. Enhanced Organizer Controls
- Integrated judging mode selection directly in track configuration
- Added visual indicators for track judging modes (badges)
- Implemented automatic criteria updates when judging mode changes

### 4. Data Model Updates
- Extended `ScoringCriterion` type to support choice-based criteria
- Added type-aware rendering for different criteria types
- Updated record formats to store investment decisions correctly

### 5. UX Improvements
- Added clear warnings when changing judging modes
- Created visual differentiation between traditional and investment tracks
- Improved feedback for judges making investment decisions

## Technical Details
- Used type field on criteria to distinguish between "numeric", "choice", and "scale"
- Extended the existing scoring system to handle non-numeric values
- Maintained backward compatibility with existing scoring data

## To-Do Before Merge
- [ ] Complete end-to-end testing of both judging modes
- [ ] Verify role labels are properly coordinated with judging modes
- [ ] Test edge cases (switching between modes, multiple tracks with different modes)
- [ ] Ensure correct display of investment decisions in scoring summaries
- [ ] Verify export/reporting functionality works with both judging types
- [ ] Check for any UI inconsistencies or display issues

## Screenshots
CleanShot 2025-05-17 at 08 51 54@2x

CleanShot 2025-05-17 at 08 51 47@2x

### CODE REVIEW

**Review by MatthewLaw1 — COMMENTED** [review]
I would suggest from a sys design perspective that for each iterative feature addition, we don't overhaul routes, but have a modifiable form field on the frontend. the codebase is getting extremely large to an unnecessary degree due to consistent additions of subdirs for independent routing on custom roles, which is not ideal. custom form fields with modifiable front end would be optimal for two reasons: 1. the implementation for custom user-added events has to be completed anyways, this brings us closer, 2. the current implementation makes post data processing difficult due to backend complexity for collating different data-sources. From a high level, the existing user flow and file structure is bloated due to lack of universal parts which prevents both customization and clean code practices, would suggest aggressive pruning urgently which can probably lead to over 7000 deletions.

**Comment by MatthewLaw1** [comment]
The sys approach should just be the wrapper and implemented the custom metadata like user roles. Suggest a likely overhaul of the supabase schema to support this outside, as both hackathons and other events have enough variance both between each other but between as well. this will be semi-necessary for future events notably with a hard deadline in june, adding bespoke solutions for individual events is digging us into a deeper hole which will require major refactoring later.

**Comment by iGotsIt** [comment]
> The sys approach should just be the wrapper and implemented the custom metadata like user roles. Suggest a likely overhaul of the supabase schema to support this outside, as both hackathons and other events have enough variance both between each other but between as well. this will be semi-necessary for future events notably with a hard deadline in june, adding bespoke solutions for individual events is digging us into a deeper hole which will require major refactoring later.

Are you talking about the organizer options or the new judging criteria or both?

**Comment by iGotsIt** [comment]
We're going to merge this now so it's live, but we do need to refactor these features in an upcoming PR.

File: components/dashboard/judge-dashboard.tsx [hunk 1]
```diff
@@ -218,9 +238,13 @@ export function JudgeDashboard({ eventId }: JudgeDashboardProps) {
     return acc;
   }, {} as Record<string, Project[]>);
 
+  console.log(projectsByTrack);
```
Comment (cyu60) [inline components/dashboard/judge-dashboard.tsx:241]: Remove log

Comment (iGotsIt) [reply components/dashboard/judge-dashboard.tsx]: fixed

**Review by iGotsIt — APPROVED** [review]
No build issues
